networkx/convert.py

```diff
@@ -70,6 +71,8 @@ def to_networkx_graph(data, create_using=None, multigraph_input=False):
         a multigraph from a multigraph.
 
     """
+    if hasattr(data, "__networkx_backend__"):
+        data = nx.utils.backends.convert_to_nx(data)
```

Comment:
I do wonder whether this block should be below the `# NX graph` block. The `# NX graph` block should also work _iff_ the backend graph correctly implements `G.adj`, `G.nodes`, and `G.graph`. In practice, for backend graphs like GraphBLAS and nx-cugraph, I wouldn't expect much performance difference, and the current implementation seems cleaner and safer to me.